app/services/tag_labeling_service.py

```
@@ -49,55 +54,66 @@ def assign_tag_to_message(self, subject_id, message, tags) -> MessageTagAssignme
+
+                tag_codes = self.extract_tag_codes_array_from_json_str(response_text)
+
+                # 최종 tag_codes가 모두 문자열인지, 빈 배열이 아닌지 확인
+                if isinstance(tag_codes, list) and all(isinstance(tc, str) for tc in tag_codes) and tag_codes:
+                    return MessageTagAssignment(subject_id=subject_id, tag_codes=tag_codes)
+                else:
+                    raise Exception("추출된 결과가 유효한 문자열 배열이 아닙니다.")
+            except Exception as e:
+                logger.error(f"LLM 응답 파싱 실패 for subject_id {subject_id} on attempt {attempt}: {e}")
+                if attempt == max_attempts:
+                    raise Exception(f"assign_tag_to_message 실패: {subject_id}에 대해 {max_attempts}번 시도했으나 실패했습니다. ({e})")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)